decisions/0002-secure-JSON-output.md

```diff
@@ -0,0 +1,84 @@
+the JSON output from Form Builder, and then generates the complaint in Optics. This
+adapter will live outside the Form Builder infrastructure. In the future, the hope
+is that teams will create their own adapters in order to integrate Form Builder
+with various external systems.
+
+This ADR is regarding how to secure this JSON output from Form Builder to the
+endpoint set by the user.
+
+Threats
+- Payload may be intercepted, exposing sensitive user answers.
+- Payload may be modified in transit, with potentially damaging consequences.
+
+Risks
+- Reputational damage due to data breaches
+- Security threats from modified payloads
+
+Mistakes from the User
+- Users could potentially enter an incorrect endpoint, which would deliver sensitive
+data to an incorrect endpoint.
+- Users could share security information accidentally, allowing malicious actors
+to receive payloads.
+
+## Proposed Decision
+
+### TLS over HTTP
+Use TLS over HTTP (HTTPS). This would ensure
+encrypted data in transit, which [satisfies MOJ standards](https://ministryofjustice.github.io/security-guidance/standards/cryptography/#cryptography).
+We came to the conclusion that mutual TLS doesn’t really bring any benefits and
+involves way too much management overhead in this context, so agreed on normal
+TLS + payload encryption with a pre-shared secret.
+
+### Encrypted JSON Payload
+Encryption of the payload is possible with JWE (JOSE toolkit). This would require
+decryption at the other end. Payload encryption is preferred over TLS alone as TLS
+is often terminated at the edge of a large network with ommunications travelling in the clear
+inside that network. Given the sensitivity of the submitted data for some forms
+and that it may need to be relied on in court, we need more confidence that only
+the intended recipient can read the data and that other actors on that network
+can’t impersonate FB by sending other requests to the endpoint.
+
+### Shared Secret
+The shared secret will be set as an ENV var in the publisher to be consumed by
+the form's runner instance. The shared secret will be used to encrypt / decrypt
+the payload.
+
+### Certificates vs Shared Secret
+Both certificates and shared secrets can be used for signing and de-serialising the payload.
+Given users may not be technical, a shared secret would be preferred as requiring users
+to generate and upload certificates may be too much to ask.
+
+
+### Overview of Solution
+
+Form Builder:
+- Connects with adapter via HTTPS using a ruby library such as [Net:HTTP](https://ruby-doc.org/stdlib-2.6.3/libdoc/net/http/rdoc/Net/HTTP.html).
+- Connection with TLS using `Net::HTTP` gem or similar.
+- Encrypts JSON payload using a shared secret and JWE protocol.
+- Sends as POST request.
+
+Adapter:
+- Receives HTTPS POST request from Form Builder.
+- Decrypts JSON payload using shared secret.
+- Also verifies request header / origin to ensure it was sent from Form Builder.
```

Comment:
The payload being encrypted with the shared secret should be sufficient for knowing the request comes from FB, I'm not sure there's any value in doing anything else.